coldfront/core/project/views.py

```diff
@@ -869,8 +879,15 @@ def post(self, request, *args, **kwargs):
             allocation_form_data = allocation_form.cleaned_data['allocation']
             if '__select_all__' in allocation_form_data:
                 allocation_form_data.remove('__select_all__')
+
+            unsigned_users = []
             for form in formset:
                 user_form_data = form.cleaned_data
+
+                # recording users with unsigned user access agreements
+                if user_form_data['user_access_agreement'] == 'Unsigned':
+                    unsigned_users.append(user_form_data['username'])
+
```

Comment:
I'd add a `continue` here for safety's sake. The `pop` of `'selected'` in `ProjectAddUsersSearchResultsView` means the following code won't run, but there might be a way to `POST` directly to this view, bypassing the form.